File: C#/Merge Two Sorted Lists.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution21 {
    public ListNode MergeTwoLists(ListNode list1, ListNode list2) {
        
        ListNode head = new ListNode(0);
        ListNode dummy = head;
        
        while(list1!=null&&list2!=null)
        {
            if(list1.val<=list2.val)
            {
                head.next = list1;
                list1 = list1.next;
            }
            else
            {
                head.next = list2;
                list2 = list2.next;
            }
            head = head.next;
        }
        
        head.next = list1 ?? list2;
        
        return dummy.next;
        
    }
}

File: Golang/Merge Two Sorted Lists.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func mergeTwoLists(list1 *ListNode, list2 *ListNode) *ListNode {
    dummyNode := &ListNode{}
    cur := dummyNode

    for list1 != nil && list2 != nil {
        if list1.Val > list2.Val {
            cur.Next = list2
            list2 = list2.Next
        } else {
            cur.Next = list1
            list1 = list1.Next
        }
        cur = cur.Next
    }

    if list1 != nil {
        cur.Next = list1
    } else if list2 != nil{
        cur.Next = list2
    }

    return dummyNode.Next
}